rio_tiler/io/rasterio.py

```diff
@@ -521,17 +521,17 @@ def feature(
             rio_tiler.models.ImageData: ImageData instance with data, mask and input spatial info.
 
         """
+        shape = _validate_shape_input(shape)
+
         if not dst_crs:
             dst_crs = shape_crs
 
         # Get BBOX of the polygon
         bbox = featureBounds(shape)
 
-        cutline = create_cutline(self.dataset, shape, geometry_crs=shape_crs)
```

Comment:
we should also make sure that `img.statistics()` return the same result with and without this. 

without the cutline handled in GDAL we now rely only on the masked array. (I think it should be fine, but better test anyway)

Reply:
There are some difference, so I'll need to check the `ImageData.statistics` method
<img width="822" alt="Screenshot 2023-05-02 at 11 32 54 AM" src="https://user-images.githubusercontent.com/10407788/235631507-ad2a64f2-2333-4247-8647-8f2d90b8821a.png">

Reply:
```python
# Using New feature method 
with Reader(src_path) as f:
    img_f = f.feature(feat)
    
# Using GDAL cutline 
with Reader(src_path) as f:
    cutline = create_cutline(f.dataset, feat, geometry_crs="epsg:4326")
    bbox = featureBounds(feat)
    img = f.part(bbox, vrt_options={"cutline": cutline})  

numpy.testing.assert_array_equal(img.mask, img_f.mask)
```

adding this kind of test could be nice (if not there yet), just to make sure we do not break anything.
The `img.data` won't be equal with the new version because we rely only on the masked array

Reply:
I would not expect this to be exactly the same, unless there is not projection going on. So if the dataset has the same projection as the feature and the `dst_crs` is not changed. Otherwise there might be differences from rasterizing the geometry in the projection of the dataset (vrt cutline) vs in the dst_crs (rasterize).

Reply:
Turns out there is a slight difference between using GDAL cutline and creating the mask using rasterize. It's definitely more `precise` with the rasterized mask (I guess this is due to reprojection of the cutline within warped vrt)

![2023-05-02 17-57-15 2023-05-02 17_57_39](https://user-images.githubusercontent.com/10407788/235729107-43333879-219c-43ab-b1af-0b0a5bf0899e.gif)


The `img.statistics()` will give different results but that's fine 👍